src/color.rs

```diff
@@ -174,6 +174,22 @@ impl<T: Primitive + 'static> Pixel for $ident<T> {
         }
     }
 
+    fn map_without_alpha<F>(&mut self, f: F) -> Self 
```

Comment:
After the conclusion in #956 these should be default implementations in the trait in `src/buffer.rs` instead, right?

Reply:
Also `(*self).clone()` can be simply `*self`, since `trait Pixel: Copy + Clone`.

Reply:
And `map_without_alpha` doesn't require `&mut self` but `&self` suffices.

Reply:
Why does making `*self` only work instead of `(*self).clone()`?

Reply:
Because of the marker trait `Copy` which allows the compiler to make bytewise copies without invoking any additional code. It will then always copy the value instead of moving it from the place. This allows simply reading the value through a reference (because it will leave the original value unmoved), instead of requiring cloning it.

Reply:
Ohhhh, haha I forgot what Copy meant. Thanks